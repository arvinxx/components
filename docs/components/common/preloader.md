---
title: Preloader
order: 2
---

# Preloader

## 图形一览

### Square

<code src="./examples/Preloader/Square" />
