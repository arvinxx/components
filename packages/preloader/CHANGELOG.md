# Changelog

## @arvinxu/preloader [1.0.2](https://github.com/arvinxx/components/compare/@arvinxu/preloader@1.0.1...@arvinxu/preloader@1.0.2) (2021-03-20)


### 👷 构建系统

* 修正构建配置问题 ([d232d17](https://github.com/arvinxx/components/commit/d232d17))

## @arvinxu/preloader [1.0.1](https://github.com/arvinxx/components/compare/@arvinxu/preloader@1.0.0...@arvinxu/preloader@1.0.1) (2021-03-20)


### 👷 构建系统

* 修改构建配置 ([e827348](https://github.com/arvinxx/components/commit/e827348))

# @arvinxu/preloader 1.0.0 (2021-03-20)


### 🎫 杂项

* 初始化 Preloader 组件 ([fa12d88](https://github.com/arvinxx/components/commit/fa12d88))


### 👷 构建系统

* 修改构建配置 ([5d6552e](https://github.com/arvinxx/components/commit/5d6552e))
