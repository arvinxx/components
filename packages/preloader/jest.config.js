const base = require('../../jest.config.base');

const packageName = '@arvinxu/preloader';

const root = '<rootDir>/packages/preloader';

module.exports = {
  ...base,
  rootDir: '../..',
  roots: [root],
  name: packageName,
  displayName: packageName,
  collectCoverageFrom: [`${root}/src/**/*.tsx`, `${root}/src/**/*.ts`],
};
