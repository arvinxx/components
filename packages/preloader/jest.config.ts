import base from '../../jest.config.base';
import { Config } from '@umijs/test';

const packageName = '@arvinxu/preloader';

const root = '<rootDir>/packages/preloader';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
  collectCoverageFrom: [`${root}/src/**/*.tsx`, `${root}/src/**/*.ts`],
};

export default config;
