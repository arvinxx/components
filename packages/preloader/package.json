{
  "name": "@arvinxu/preloader",
  "version": "1.0.2",
  "files": [
    "dist"
  ],
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "sideEffects": [
    "*.less"
  ],
  "homepage": "https://github.com/arvinxx/components/tree/master/packages/preloader#readme",
  "repository": "git+https://github.com/arvinxx/components.git",
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "scripts": {
    "build": "father build",
    "test": "jest",
    "test:update": "jest -u",
    "prepublishOnly": "pnpm run build",
    "cov": "jest --coverage",
    "clean": "rm -rf es lib dist build coverage .umi"
  }
}
