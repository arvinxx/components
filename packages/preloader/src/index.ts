export { default as Square } from './Square';
