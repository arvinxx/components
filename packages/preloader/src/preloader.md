---
title: Preloader
order: 1
group: 信息展示
---

# Preloader

[![NPM version][version-image]][version-url] [![NPM downloads][download-image]][download-url]

<!-- npm url -->

[version-image]: http://img.shields.io/npm/v/@arvinxu/preloader.svg?color=deepgreen&label=latest
[version-url]: http://npmjs.org/package/@arvinxu/preloader
[download-image]: https://img.shields.io/npm/dm/@arvinxu/preloader.svg
[download-url]: https://npmjs.org/package/@arvinxu/preloader

## 图形一览

### Square

<code src="./demos/Square" ></code>
