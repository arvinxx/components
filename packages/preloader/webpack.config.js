const path = require('path');
const config = require('../../webpack.config');

module.exports = {
  ...config,
  output: {
    ...config.output,
    library: 'preloader',
    path: path.resolve(__dirname, 'dist'),
  },
};
