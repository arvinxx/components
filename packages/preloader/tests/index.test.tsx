import React from 'react';
import { render } from '@testing-library/react';

import { Square } from '@arvinxu/preloader';

describe('Preloader', () => {
  it('Square', () => {
    const { container } = render(<Square />);
    expect(container).toMatchSnapshot();
  });
});
