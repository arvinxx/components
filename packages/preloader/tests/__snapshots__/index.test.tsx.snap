// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader Square 1`] = `
<div>
  <div
    class="avx-preloader-square-container"
  >
    <div
      class="avx-preloader-square-item"
      style="background: rgb(24, 144, 255);"
    />
    <div
      class="avx-preloader-square-item avx-preloader-square-animation-delay"
      style="background: rgb(24, 144, 255);"
    />
    <div
      class="avx-preloader-square-item avx-preloader-square-animation-delay"
      style="background: rgb(24, 144, 255);"
    />
    <div
      class="avx-preloader-square-item"
      style="background: rgb(24, 144, 255);"
    />
  </div>
</div>
`;
